src/utils/draftbots.js

```diff
@@ -104,29 +104,47 @@ const scaleSimilarity = (value) =>
 const SYNERGY_SCALE = 0.2;
 const MAX_SYNERGY = 10;
 
-let synergyMatrix;
+const xorStrings = (a, b) => {
+  let s = '';
 
-export const setSynergyMatrix = (size) => {
-  synergyMatrix = [];
-  for (let i = 0; i <= size; i++) {
-    synergyMatrix.push(new Array(size + 1).fill(null));
+  // use the longer of the two words to calculate the length of the result
+  for (let i = 0; i < Math.max(a.length, b.length); i++) {
+    // append the result of the char from the code-point that results from
+    // XORing the char codes (or 0 if one string is too short)
+
+    // eslint-disable-next-line no-bitwise
+    s += String.fromCharCode((a.charCodeAt(i) || 0) ^ (b.charCodeAt(i) || 0));
   }
+
+  return s;
 };
 
-export const getSynergy = (index1, index2, synergies) => {
-  if (!synergies) return 0;
-  if (synergyMatrix[index1][index2] === null) {
-    const similarityValue = similarity(synergies[index1], synergies[index2]);
+const synergyCache = {};
+
+export const getSynergy = (card1, card2) => {
+  if (
```

Comment:
This can be simplified to `card1?.details?.embedding?.length && card2?.details.embedding?.length` Same in the other places you use this.